composer/core/state.py

```diff
@@ -468,13 +536,32 @@ def __init__(
         self.profiler: Optional[Profiler] = None
 
         self.deepspeed_config = deepspeed_config
-        self.fsdp_config = fsdp_config
-        self.fsdp_auto_wrap = fsdp_auto_wrap
+        parallelism_config = parallelism_config or {}
+        self.fsdp_config = parallelism_config.get('fsdp', None)
+        self.tp_config = parallelism_config.get('tp', None)
+        if self.fsdp_config is not None:
+            from composer.distributed import patch_pytorch
+
+            # Add an earlier call to patch_pytorch as we require device_mesh slicing before any
+            # model wrapping.
+            patch_pytorch()
+
+        if self.tp_config is not None:
+            if version.parse(torch.__version__.split('.dev')[0]) < version.parse('2.3.0'):
+                raise ValueError('Tensor parallelism (TP) requires torch>=2.3.0.')
+            if self.fsdp_config is None:
```

Comment:
I have tweaked the error message and will leave as is. unfortunately, it seems currently the default values don't work since we require dp_shard>=2 :(

We will change to default on once torch issues are resolved